consensus/src/sync/peer_list.rs

```diff
@@ -9,6 +10,7 @@ use nimiq_network_interface::network::Network;
 pub struct PeerList<N: Network> {
     peers_set: HashSet<N::PeerId>,
     peers: Vec<N::PeerId>,
+    notify_nonempty: Arc<Notify>,
```

Comment:
This could probably do with some comment. I am also not too happy about the name, but right now have no better suggestion.

Reply:
Added a comment.